synfig-studio/src/gui/instance.cpp

```diff
@@ -1631,26 +1631,63 @@ void
 Instance::gather_uri(std::set<synfig::String> &x, const synfig::Layer::Handle &layer) const
 {
 	if (!layer || !layer->get_canvas()) return;
+	int count =0; //will be used to count layers in the group
+	
+	//check if the layer is group layer
+	ParamVocab vocab = layer->get_param_vocab();
+	synfig::Layer::Handle layerfinal,lay;
+	bool some_bool = false;
+	for(ParamVocab::const_iterator i = vocab.begin(); i != vocab.end(); ++i)
+	{
+		if(i->get_description() == "Group content")
+		{
+			etl::handle<synfig::Layer_PasteCanvas> p = etl::handle<synfig::Layer_PasteCanvas>::cast_dynamic(layer);
+			synfig::Canvas::Handle canvas = p->get_sub_canvas();
+			if(canvas)
+			{
+				for(IndependentContext i = canvas->get_independent_context(); *i; i++)
+				{
+					lay = (*i);
+					count++;
+				}
+				if(count==1)
+				{
+					some_bool = true;//yes only one layer 
+				}				
+			}
+		}
+	}
 
 	FileSystem::Handle file_system = layer->get_canvas()->get_file_system();
 	if (!file_system) return;
 
-	ParamVocab vocab = layer->get_param_vocab();
+	//if yes then the layer inside group should be processed not the group!
+	if(some_bool)
+	{
+		vocab = lay->get_param_vocab();
+		layerfinal = lay;
+	}
+	else
+	{
+		layerfinal = layer;
```

Comment:
I suggest to use "layer" instead of "layerfinal". Then, this code will be enough:

```
if (count == 1){
		vocab = lay->get_param_vocab();
		layer = lay;
	}

Reply:
I guess it didn't worked due to being const

Reply:
Ah, I understand! ^__^